### DISCUSSION TITLE
Reset password via SDK

### DISCUSSION THREAD

**Discussion by norricorp** [directus/directus#17826] [Archive]
Am I right in thinking that the reset password api can't be used directly?
That the token parameter is NOT the access token from authentication but is a new token from request a password change? 

Would a useful new feature allow a user, via a front end, to be able to change password whilst logged in. The parameters might be the access token, new password or email, old pw, new pw.

The email token route is great for fogotten passwords but once logged in, having to receive an email seems unnecessary.

**Comment by ctholho** [comment]
[Updating the current user](https://docs.directus.io/reference/system/users.html#update-the-current-user) is possible, given that you have edited the role permissions accordingly.

Make sure that when you edit the permissions, you allow users to only change their own password.
<img width="1160" alt="Screenshot 2023-03-21 at 14 37 49" src="https://user-images.githubusercontent.com/28910105/226623158-63338856-e397-453b-b201-84c97e9b7fd8.png">

This is handled automatically by the minimum required settings for Directus App Access. If your users don't have access to the app, you must set the permission yourself.

**Reply by norricorp** [reply]
Hi @ctholho,
Thanks for the reply. I was just setting the entire user record to editable (just to see it working) so this is useful.
But the token that the API requires, is that the access token generated at authentication or is it unique when a password request email is created?

The problem I am having with request password reset API where I give two params, email and alternative URL (ie doesn't go to directus but other website). I get an error of

Uncaught (in promise) Error: Url "http://localhost:3000/resetpw" can't be used to reset passwords.

I have searched thru' the directus repo on GitHub and not found that error so no idea where it's coming from.

**Reply by ctholho** [reply]
You don't need a token if you use [Updating the current user](https://docs.directus.io/reference/system/users.html#update-the-current-user) or [with the sdk](https://docs.directus.io/reference/sdk.html#update-the-current-users) and there won't be an email.

**Reply by norricorp** [reply]
So don't use password.reset?

**Reply by borisovdev** [reply]
@norricorp, have you tried to use "PASSWORD_RESET_URL_ALLOW_LIST" environment variable?
https://docs.directus.io/self-hosted/config-options.html#security

**Comment by alpha-mo** [comment]
> @norricorp, have you tried to use "PASSWORD_RESET_URL_ALLOW_LIST" environment variable? https://docs.directus.io/self-hosted/config-options.html#security

I would love to see an example of this cause I'm having a hard time fixing this

**Comment by norricorp** [comment]
It is a while since I looked at this. 
I didn't use "PASSWORD_RESET_URL_ALLOW_LIST" 
The approach I took for forgotten password was to tell directus to send a request to the frontend with a token
       directus.auth.password.request(email, import.meta.env.VITE_SVELTE_URL + "/resetpw");
resetpw allowed the user to enter a new password which then
       await directus.auth.password.reset( token, new_password )
       
For change password, put up form, then
      await directus.auth.login({ email: $emailName, password: old_password }) 
followed by
      await directus.users.me.update({ password: new_password })

This was using release 10 of the sdk. I returned to directus (using sveltekit rather than svelte frontend) the last few weeks and used sdk 15 and functions have been rewritten so syntax is now very different
